gematria/granite/graph_builder_model_inference.cc

```diff
@@ -346,7 +424,48 @@ GraphBuilderModelInference::FromTfLiteModel(
   }
   llvm::Expected<std::unique_ptr<tflite::Interpreter>> interpreter =
       CreateInterpreter(*tflite_model);
-  if (auto error = interpreter.takeError()) return error;
+  if (llvm::Error error = interpreter.takeError()) return error;
+
+  // Get a mapping between the input tensors used in the model and their
+  // corresponding indices.
+  auto input_tensor_to_idx =
+      std::make_unique<std::vector<int>>(kNumPossibleInputTensors, -1);
+  for (int idx : (*interpreter)->inputs()) {
+    const std::string_view name = (*interpreter)->GetInputName(idx);
+    input_tensor_to_idx->at(static_cast<int>(GetInputTensorFromName(name))) =
+        idx;
+  }
+
+  // Checks whether the model has all input tensors required to have for all
+  // graph builder models.
+  std::vector<InputTensor> missing_inputs;
+  for (const InputTensor input : kRequiredInputTensors) {
+    if (input_tensor_to_idx->at(static_cast<int>(input)) == -1)
+      missing_inputs.push_back(input);
+  }
```

Comment:
The approach I went for was to make sure the required input tensors were there, and infer the configuration (i.e. `uses_deltas` or not from the presence of the delta block index tensor). So if `kDeltaBlockIndexTensor` is missing. we just assume it is a `seq2num` model and continue. Maybe it would be better to log any such assumptions we make about the configuration, e.g.
```
Found `ModelBase.delta_block_index_tensor`, assuming model uses deltas.
```
Or would that be unnecessarily verbose?